### ISSUE TITLE
Can't get camera api to run

### ISSUE THREAD

**Issue by paulsonnentag** [janpaul123/paperprograms#55]
I'm not sure what I'm doing wrong, but the new camera api is not working for me. 

This is my paper program:
```JavaScript
// Cam
importScripts('paper.js');

(async () => {
  var canvas = await paper.get('canvas');
  var ctx = canvas.getContext('2d');

  setInterval(async () => {
    const papers = await paper.get('papers'); 
    const camera = await paper.get('camera');

    paper.drawFromCamera(ctx, camera, papers[2068].points, papers[757].points);
    ctx.commit();
  });
})();
```

Initially I got the error that the data is not transferable:

<img width="984" alt="screen shot 2018-05-01 at 18 26 35" src="https://user-images.githubusercontent.com/650540/39481951-843680ac-4d6d-11e8-8e15-10a5bd99b9e0.png">

I'm wondering why [this line](https://github.com/janpaul123/paperprograms/blob/master/client/projector/ProjectorMain.js#L25) doesn't have an await even though [ImageCapture.grabFrame](https://developer.mozilla.org/en-US/docs/Web/API/ImageCapture/grabFrame) returns a Promise. Is that a mistake or am I missing something?

When I add a await I get a different Error from `ImageCapture.grabFrame`:

<img width="987" alt="screen shot 2018-05-01 at 18 28 14" src="https://user-images.githubusercontent.com/650540/39481949-81cd910c-4d6d-11e8-8f16-cdff86ba110a.png">

@shaunlebron  Can you give me some pointers what I'm doing wrong. Thanks, I'm really excited to build something with the new camer API.

**Comment by shaunlebron** [comment]
Thanks, there were some changes made before merge that I probably didn't check?  Will look tonight

**Comment by shaunlebron** [comment]
Good catch on the `grabFrame` issue.  Fixed that here: #56 

As for the last error about the `Track` being in an invalid state, I couldn't reproduce.  Maybe the camera access is blocked?  Here's where the projector page retrieves the video track of the camera:

https://github.com/janpaul123/paperprograms/blob/a35abb97008d711c29e5092c9b99117529c02037/client/projector/ProjectorMain.js#L15-L22

**Comment by paulsonnentag** [comment]
I still haven't fully resolved this issue. I could find out what's behind the "invalid state" error. According to the [spec](https://w3c.github.io/mediacapture-image/index.html#dom-imagecapture-grabframe) this error occurs if the `readyState` of the `track` is not "live". I couldn't figure out why this occurred though. Once I started logging the readyState of the track the problem no longer occurred. It seems that the implementation of Chrome is still buggy. Right now it works for me if I reload the projector from time to time.

Another issue that I've noticed is that the call to `this.props.grabCameraImageAndProjectionData()` would sometimes reject with `undefined`.

<img width="989" alt="screen shot 2018-05-02 at 20 27 22" src="https://user-images.githubusercontent.com/650540/39541869-4a6f93f6-4e47-11e8-9638-be043da20700.png">

This error didn't affect the program immediately because I call `get('camera')` continuously with setInterval but long term this will probably lead to a memory leak:

```JavaScript
  setInterval(async () => {
    const papers = await paper.get('papers');
    const camera = await paper.get('camera'); // if the call on the projector side fails this will block forever

     ....
   }, 100)
```

Has someone else experienced these problems?
